rpcs3/Emu/RSX/GL/rsx_gl_texture.cpp

```diff
@@ -89,6 +90,84 @@ namespace rsx
 			return 1.0f;
 		}
 
+		u16 texture::get_pitch_modifier(u32 format)
+		{
+			switch (format)
+			{
+			case CELL_GCM_TEXTURE_COMPRESSED_HILO8:
+			case CELL_GCM_TEXTURE_COMPRESSED_HILO_S8:
+			default:
+				LOG_ERROR(RSX, "Unimplemented Texture format : 0x%x", format);
+				return 0;
+			case ~(CELL_GCM_TEXTURE_LN | CELL_GCM_TEXTURE_UN) & CELL_GCM_TEXTURE_COMPRESSED_B8R8_G8R8:
+			case ~(CELL_GCM_TEXTURE_LN | CELL_GCM_TEXTURE_UN) & CELL_GCM_TEXTURE_COMPRESSED_R8B8_R8G8:
+				return 4;
+			case CELL_GCM_TEXTURE_B8:
+				return 1;
+			case CELL_GCM_TEXTURE_COMPRESSED_DXT1:
+			case CELL_GCM_TEXTURE_COMPRESSED_DXT23:
+			case CELL_GCM_TEXTURE_COMPRESSED_DXT45:
+				return 0;
+			case CELL_GCM_TEXTURE_A1R5G5B5:
+			case CELL_GCM_TEXTURE_A4R4G4B4:
+			case CELL_GCM_TEXTURE_R5G6B5:
+			case CELL_GCM_TEXTURE_G8B8:
+			case CELL_GCM_TEXTURE_R6G5B5:
+			case CELL_GCM_TEXTURE_DEPTH16:
+			case CELL_GCM_TEXTURE_DEPTH16_FLOAT:
+			case CELL_GCM_TEXTURE_X16:
+			case CELL_GCM_TEXTURE_R5G5B5A1:
+			case CELL_GCM_TEXTURE_D1R5G5B5:
+				return 2;
+			case CELL_GCM_TEXTURE_A8R8G8B8:
+			case CELL_GCM_TEXTURE_X32_FLOAT:
+			case CELL_GCM_TEXTURE_Y16_X16_FLOAT:
+			case CELL_GCM_TEXTURE_D8R8G8B8:
+			case CELL_GCM_TEXTURE_COMPRESSED_B8R8_G8R8:
+			case CELL_GCM_TEXTURE_COMPRESSED_R8B8_R8G8:
+			case CELL_GCM_TEXTURE_DEPTH24_D8:
+			case CELL_GCM_TEXTURE_DEPTH24_D8_FLOAT:
+			case CELL_GCM_TEXTURE_Y16_X16:
+				return 4;
+			case CELL_GCM_TEXTURE_W16_Z16_Y16_X16_FLOAT:
+				return 8;
+			case CELL_GCM_TEXTURE_W32_Z32_Y32_X32_FLOAT:
+				return 16;
+			}
+		}
+
+		bool texture::mandates_expansion(u32 format)
+		{
+			switch (format)
+			{
+			default:
+				return false;
+			case CELL_GCM_TEXTURE_A1R5G5B5:
+				return true;
+			case CELL_GCM_TEXTURE_A4R4G4B4:
```

Comment:
If I remember correctly it requires expansion because there is no 1-5-5-5 storage format in gfx api. The closest is 5-5-5-1 but it requires manually moving the bits. Same for R6G5B5 ; The closest is R5G6B5 but it requires some bit manipulation too. On the other hand, R5G6B5 is supported, it only requires some swizzling.